benchmark/benchmark_scenarios.py

```diff
@@ -0,0 +1,55 @@
+from twisted.internet.defer import maybeDeferred, Deferred, succeed
+
+
+class RunningScenario:
+
+    def __init__(self):
+        self.scenario_established = Deferred()
+        self.scenario_maintained = Deferred()
+
+    def established(self):
+        """
+        :return: A Deferred that fires when the desired scenario is
+            established (e.g. that a certain load is being applied).
+        """
+        return self.scenario_established
+
+    def maintained(self):
+        """
+        :return: A Deferred that fires with an errback id the desired
+            scenario fails to hold between being established and being
+            stopped.  This Deferred never fires with a callback.
+        """
+        return self.scenario_maintained
+
+    def stop(self):
+        """
+        Stop the scenario from being maintained.
+
+        :return: A Deferred that fires when the desired scenario is
+            stopped.
+        """
+        return succeed(None)
+
+
+class _NoLoadScenario(object):
+
+    def __init__(self, clock, client):
+        self.clock = clock
+        self.client = client
+
+    def start(self):
+        running = RunningScenario()
+        running.established().callback(None)  # no setup needed
+        return running
```

Comment:
I just got rather confused when reading the tests. There's a thing called a "scenario" which just has a `start` method, and there's another thing called a "running scenario" which isn't a scenario but actually has a large chunk of the logic for a scenario? Is that right?

I'd strongly consider adding interfaces here, and maybe changing the names (although I don't have any good ideas right now). At least, please provide a module docstring that explains the core concepts.